python/lsst/ts/externalscripts/maintel/tma/random_walk_and_take_image_gencam.py

```diff
@@ -0,0 +1,583 @@
+                  maximize the time on target. Other options are; 1-noUnWrap, 2-optimize.
+                type: integer
+                minimum: 1
+                maximum: 3
+                default: 3
+              camera_sal_indexes:
+                description: SAL Indexes for the Generic Cameras.
+                type: array
+                items:
+                  type: integer
+                  minimum: 0
+                  maximum: 2147483647
+              exp_times:
+                description: Exposure times used on each camera.
+                type: array
+                items:
+                  type: number
+                  minimum: 0
+              num_exp:
+                description: Number of exposures for all cameras.
+                type: integer
+                minimum: 1
+                default: 1
+              sleep_time:
+                description: Sleep time between exposures.
+                type: number
+                minimum: 0
+                default: 0
+              reason:
+                description: Optional reason for taking the data.
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: null
+              program:
+                description: Optional name of the program this data belongs to, e.g. WFD, DD, etc.
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: null
+              camera_playlist:
+                description: >-
+                  Optional name a camera playlist to load before running the script.
+                  This parameter is mostly designed to use for integration tests and is
+                  switched off by default (e.g. null).
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: null
+              max_dome_mount_az:
+                description: >-
+                  Maximum difference between dome position and target mount azimuth in degrees.
+                  If the actuall difference is larger than this parameter, the dome moves.
+                type: number
+                default: 10.
+              dome_offset:
+                description: Artificial offset for when the dome loses reference.
+                type: number
+                default: 0
+            required:
+            - camera_sal_indexes
+            - exp_times
+            - total_time
+        """
+        return yaml.safe_load(schema_yaml)
+
+    async def load_playlist(self):
+        """Load playlist."""
+        raise NotImplementedError()
+
+    async def move_dome(self, az: float):
+        """Move the dome to a give position
+
+        Parameters
+        ----------
+        az : float
+            New dome Azimuth position in degrees.
+        """
+        await self.tcs.rem.mtdome.cmd_exitFault.set_start()
+        await asyncio.sleep(5)
```

Comment:
If you can get rid of this delay, please do so. If not, please add a code comment explaining why it is necessary.